Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIDeclarativeNetRequestCocoa.mm

```diff
@@ -76,7 +76,7 @@
     }
 
     for (NSString *savedIdentifier in savedRulesetState) {
-        auto ruleset = extension().declarativeNetRequestRuleset(savedIdentifier);
+        auto ruleset = protectedExtension()->declarativeNetRequestRuleset(savedIdentifier);
```

Comment:
We should cache the protected extension before the loop to avoid repeated calls to `protectedExtension()`.

Reply:
Fixed!